indico/htdocs/js/indico/modules/registration/form/field.js

```diff
@@ -900,7 +891,11 @@ ndRegForm.directive('ndAccommodationField', function(url) {
             });
 
             scope.$watch('accommodation.choice', function(newValue) {
-                updateAccommodationPostData('choice', newValue);
+                if (!newValue) {
+                    $('[name=field_{0}]'.format(scope.field.id)).val('{}');
```

Comment:
Can we please put something referencing a close parent element of the registration form here? An attribute selector over the whole document is pretty ugly.